ambry-network/src/main/java/com.github.ambry.network/Transmission.java

```diff
@@ -132,8 +132,7 @@ public SocketAddress getRemoteSocketAddress() {
   /**
    * Close the connection for the socket channel
    */
-  public abstract void close()
-      throws IOException;
```

Comment:
because the doc of this method says "Close the connection for the socket channel" and socket channel closes usually throw `IOException`. If you look at the code of `PlainTextTransmission`, it is catching the `IOException` when it doesn't need to. The `Selector` will eventually catch it (code for which is also there and you have removed now) because that is the one that cannot throw anymore (because `Selectable::close()` doesn't allow that (and it should in my opinion). Channel closes always throw `IOException`.